stl/inc/xstring

```diff
@@ -2541,6 +2587,17 @@ public:
             _Ptr                    = _Unfancy(_Fancyptr);
             _Construct_in_place(_My_data._Bx._Ptr, _Fancyptr);
         }
+#ifdef __cpp_lib_constexpr_string
+        else if (_STD is_constant_evaluated()) { // SSO disabled in constexpr context
+            const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
+            _Ptr                    = _Unfancy(_Fancyptr);
+            _Construct_in_place(_My_data._Bx._Ptr, _Fancyptr);
+        }
+
+        if (_STD is_constant_evaluated()) { // Copying into uninitialized memory is technically UB, so fill it once
+            _Traits::assign(_Ptr, _New_capacity + 1, _Elem());
+        }

```

Comment:
Yeah, I was surprised myself, but clang was really unhappy without it

Reply:
Do you recall the types of errors you were seeing?  We should ideally try to determine whether this is a Clang bug that needs to be reported/resolved ASAP rather than discovering the unhappiness when we are finally able to turn `_CONSTEXPR20_CONTAINER` on for Clang.

Reply:
I think we need to locally remove mutable from the iterator machinery and compile again with clang

Reply:
I've verified that MSVC doesn't need any of those regions (compiles and passes tests).  Since Clang 12 should have the `mutable` fix, we should see if compiling these changes with Clang still complains about lifetime without those regions.  If it does, I believe we should report a Clang bug and `// TRANSITION` those blocks. Otherwise, we should remove them!

Reply:
@mnatsuhara determined that Clang is correct here: https://github.com/microsoft/STL/pull/1502#issuecomment-785515569